File: app/src/main/java/io/github/mohamedisoliman/androiddev/data/model/Image.java
package io.github.mohamedisoliman.androiddev.data.model;

import com.google.gson.annotations.Expose;
import java.util.List;

public class Image {

  @Expose
  private String id;
  @Expose
  private List<Resolution> resolutions;
  @Expose
  private Source source;
  @Expose
  private Variants variants;

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public List<Resolution> getResolutions() {
    return resolutions;
  }

  public void setResolutions(List<Resolution> resolutions) {
    this.resolutions = resolutions;
  }

  public Source getSource() {
    return source;
  }

  public void setSource(Source source) {
    this.source = source;
  }

  public Variants getVariants() {
    return variants;
  }

  public void setVariants(Variants variants) {
    this.variants = variants;
  }
}

File: app/src/main/java/io/github/mohamedisoliman/androiddev/data/model/Image.kt
package io.github.mohamedisoliman.androiddev.data.model

import com.google.gson.annotations.Expose

class Image {

  @Expose
  var id: String? = null
  @Expose
  var resolutions: List<Resolution>? = null
  @Expose
  var source: Source? = null
  @Expose
  var variants: Variants? = null
}
